src/java.net.http/share/classes/jdk/internal/net/http/Http1Request.java

```diff
@@ -109,18 +109,18 @@ public void collectHeaders0(StringBuilder sb) {
         final HttpHeaders uh = userHeaders;
 
         // Filter any headers from systemHeaders that are set in userHeaders
-        systemHeaders = HttpHeaders.of(systemHeaders.map(), (k,v) -> uh.firstValue(k).isEmpty());
+        final HttpHeaders sh = HttpHeaders.of(systemHeaders.map(), (k,v) -> uh.firstValue(k).isEmpty());
 
         // If we're sending this request through a tunnel,
         // then don't send any preemptive proxy-* headers that
         // the authentication filter may have saved in its
         // cache.
-        collectHeaders1(sb, systemHeaders, nocookies);
+        collectHeaders1(sb, sh, nocookies);
 
         // If we're sending this request through a tunnel,
         // don't send any user-supplied proxy-* headers
         // to the target server.
-        collectHeaders1(sb, userHeaders, nocookies);
+        collectHeaders1(sb, uh, nocookies);
 
```

Comment:
I found this part of the change confusing, could I suggest that the comment below is changed to say:

// Gather all 'Cookie:' headers from the unfiltered system headers, and the user headers
// and concatenate their values in a single line

Reply:
Good point! Done.